lib/cn/spill.h

```diff
@@ -40,7 +55,30 @@ struct cn_compaction_work;
  */
 /* MTF_MOCK */
 merr_t
-cn_spill(struct cn_compaction_work *w);
+cn_subspill(
+    struct spillctx           *sctx,
+    struct subspill           *ss,
+    struct cn_tree_node       *node,
+    uint64_t                   node_dgen,
+    void                      *ekey,
+    uint                       eklen);
+
+
+/* MTF_MOCK */
+merr_t
+cn_spill_init(struct cn_compaction_work *w, struct spillctx **sctx_out);
```

Comment:
I think our unwritten convention is to name functions `*_init()` when they initialize subsystems (e.g, `cn_tree_init()`) or objects that have already been allocated (e.g., `cn_lcur_init()`).  Functions that allocate and initialize should be named `*_create()`.

Reply:
Looking at this more, you could keep it as `_init()`, change it to not allocate the context, use local var `struct spillctx ctx`, and pass `&ctx` to `cn_spill_init()`?  Downside: expose `struct spillctx` in header file.

Reply:
Also, the subspill array may be pretty large to have on the stack.